### DISCUSSION TITLE
edrumulus MIDI output

### DISCUSSION THREAD

**Discussion by 3hhh** [corrados/edrumulus#118] [General]
I noticed that [edrumulus sends a MIDI note off message after every note](https://github.com/corrados/edrumulus/blob/7a76af578ab40a4de08a3cbebf3f445dc912fdc5/edrumulus.ino#L171-L250) and also sends an aftertouch with velocity 127 on the beginning of a choke and aftertouch 0 at the end.

What's the reasoning behind this?

Doesn't "note off" mean that the receiver should stop playing that note?
I'd usually expect that on choke only.

And "aftertouch X" was intended to change the velocity of that note to X, wasn't it? So why tell the receiver on choke on, that the output should change to maximum loudness and on choke end mute it? That seems odd.
At most I'd have expected it the other way around.

**Comment by corrados** [comment]
You can find a lot of discussions about the note off in several forums. As far as I know, the MIDI standard requires to do this.

With Edrumulus, I implemented the note off and aftertouch the same way as Roland drum modules do it.
